tests/ClientTest.php

```diff
@@ -521,6 +519,50 @@ public function testBuildEventWithDefaultValues(): void
         $client->captureEvent(Event::createEvent());
     }
 
+    public function testBuildEventDontOverwriteEventPropertiesWithDefaultValues(): void
```

Comment:
Yeah, but the scope seems to become bigger. I tried to fix the UTs but when I fix one, another falls. It seems like a lot of UTs are build and expect the values from the main scope and not from the event. So I dont know what to do, ofcourse I can change them all. But is that something that we want? Or should I revert it all, and just change the tags logic to merging. And still let the other Event fields be 'useless'?

Reply:
>Or should I revert it all, and just change the tags logic to merging

We definitely want to override the event properties only if they have not been set, or in the case of tags we always want to merge the data with the existing one.

>It seems like a lot of UTs are build and expect the values from the main scope and not from the event

I'm not sure I get what the issue is, would you mind explaining it again to me? To clarify what I meant, I think it should be possible to depend on a data provider that returns the following data for each test case:

- The configured `Options` object filled with some data for the properties we are interested in
- The `Event` object: we should handle two cases, one with the data not filled at all and one with some data filled for the properties we are interested in
- The `Event` object that should be compared against the one passed to `captureEvent()` to

Once you have the data provider in place it's just a matter of comparing the event object returned from the `captureEvent()` method with the expected object returned from the data provider. Does it makes more sense now or am I still missing something?

Reply:
> Or should I revert it all, and just change the tags logic to merging

I came here because the Event environment gets overwritten by the options, so it's definitely an issue for the other fields as well.